### ISSUE TITLE
Getting associated data from an object path obtained as an argument of a method.

### ISSUE THREAD

**Issue by mulkieran** [diwic/dbus-rs#73]
This question is somewhat related to the problem of method dispatch on multiple parameters (rather than just the first).

I am planning to have a method, m, which accepts object paths as arguments. The client has identified the appropriate object paths to pass as arguments by examining the result of GetManagedObjects() and identifying the correct object paths by matching on the interface and property values. The arguments as received by the service have been constructed as ```MessageItem::ObjectPath<some Path>```. So, while the first argument, on which the method has been dispatched, contains the actual object path with associated data, the other object path arguments do not.

It is desirable, from my point of view, to use the ```Path``` argument to obtain the corresponding ```ObjectPath``` from the tree. The reason for this is that I plan to have decorated the ObjectPath with essential data which my method requires for its operation. At this time, it is not possible to lookup an ObjectPath in the tree in order to discover its data.

Is that a fundamental limitation, or could it be removed by means of a ```get``` method on the tree?

Note: It would also be possible to have a look-aside table attached to the tree which mapped ```Path```s to the appropriate data and to choose not to decorate the ObjectPath at all. But that seems to be working a bit outside the expectations of this library and seems kind of clumsy.

**Comment by tasleson** [comment]
I went back and looked at dbus-python which I used for the lvm dbus service I wrote.  Even though internally the library must keep a table to associate a path with it's associated object, it doesn't appear to expose it.  I created and kept a hash table, to allow me to get the object that was registered to the library for the dbus service by object path.  I think it would be useful if the dbus service library exposed a method for this look up, instead of requiring the library user to duplicate this.
